crates/cli/src/main.rs

```diff
@@ -37,14 +37,19 @@ async fn main() -> std::io::Result<()> {
                 )
             })?;
 
-            let prover_vk = config.verifying_key.and_then(|s| s.try_into().ok()).and_then(
-                |vk: VerifyingKey| match vk {
-                    VerifyingKey::Ed25519(key) => Some(key),
-                    _ => None,
-                },
-            );
+            let verifying_key_algorithm = config.verifying_key_algorithm.as_str();
+
+            if verifying_key_algorithm.is_empty() {
+                return Err(std::io::Error::new(std::io::ErrorKind::InvalidInput, "verifying key algorithm is required"));
+            }
+
+            if !["ed25519", "secp256k1", "secp256r1"].contains(&verifying_key_algorithm) {
+                return Err(std::io::Error::new(std::io::ErrorKind::InvalidInput, "invalid verifying key algorithm"));
+            }
+
+            let prover_vk = VerifyingKey::from_algorithm_and_bytes(verifying_key_algorithm, config.verifying_key.unwrap().as_bytes()).unwrap();
```

Comment:
When we extend usage of `algorithm` like in this PR, we should probably consider using an enum instead of the plain string. We could, for example, convert it from a config str to an appropriate enum here, like
`CryptoAlgorithm::try_from(config.verifying_key_algorithm)`, doing the validation in that `try_from`.

Reply:
This is a great idea, thank you! Applied that

Reply:
I've pulled this out into #188 , because it also affects things I'm working on currently. It will also keep the diff a little bit smaller for your PR here. 
Solution is basically the same, with small differences:
* I used CryptoAlgorithm instead of KeyAlgorithm (In naming, "Crypto" is a superset for signatures and keys for me)
* Some corresponding changes in CryptoPayload we discussed in Slack

Reply:
I merged your PR, thank you!